sdk/workflow/executor/executor.go

```diff
@@ -83,17 +83,15 @@ func NewWorkflowExecutor(apiClient *client.APIClient) *WorkflowExecutor {
 // RegisterWorkflow Registers the workflow on the server.  Overwrites if the flag is set.  If the 'overwrite' flag is not set
 // and the workflow definition differs from the one on the server, the call will fail with response code 409
 func (e *WorkflowExecutor) RegisterWorkflow(overwrite bool, workflow *model.WorkflowDef) error {
-	response, err := e.metadataClient.RegisterWorkflowDef(
+	_, err := e.metadataClient.RegisterWorkflowDef(
 		context.Background(),
 		overwrite,
 		*workflow,
 	)
 	if err != nil {
 		return err
 	}
-	if response.StatusCode > 299 {
```

Comment:
These checks after 
```
if err != nil {
	return err
}
```

don't make sense. `err` is going to be not nil and this line won't be reached.